linera-service/src/proxy.rs

```diff
@@ -57,8 +64,29 @@ impl Proxy {
     }
 
     /// Constructs and configures the [`Proxy`] given [`ProxyOptions`].
-    fn from_options(options: ProxyOptions) -> Result<Self> {
-        let config = ValidatorServerConfig::read(&options.config_path)?;
+    async fn from_options(options: ProxyOptions) -> Result<Self> {
+        let mut tmp_config = ValidatorServerConfig::read(&options.config_path)?;
+        if options.kube {
+            let pod_name_to_ip = fetch_pod_ips().await?;
+
+            // Assumes that each service will have just one pod. Since we have one service per shard, the first pod will always have
```

Comment:
Would it make sense to have one service per validator instead, with one pod per shard? (I don't have experience with Kubernetes. :shrug:)

Reply:
We'll probably have multiple services per validator (for example a prometheus instance will run as a separate service).

Reply:
Makes sense, but should each worker have its own service, or should there just be one "workers service" with one _pod_ per worker?

Reply:
Kubernetes has a bunch of primitives we can use, so I think the answer to your question is the output of a more thorough design excercise.